### ISSUE TITLE
Applying dependancy's to github package's

### ISSUE THREAD

**Issue by Quirksmode** [jspm/registry#518]
Hi, have been using JSPM for a while now and absolutely loving it, though have run into an issue with declaring a dependancy for the ng-file-upload (https://github.com/danialfarid/ng-file-upload) package.

I managed to get ng-file-upload working nice and easily with JSPM, but I am struggling with how I can include the ng-file-upload shim as a dependancy. When I check the Network tools, the shim is not being loaded in.

System.config({
  "map": {
    "angular": "github:angular/bower-angular@1.4.3",
    "ng-file-upload": "github:danialfarid/ng-file-upload-bower@6.0.3",

    // Declare dependancy
    "github:danialfarid/ng-file-upload-bower@6.0.3": {
      "ng-file-upload-shim": "github:danialfarid/ng-file-upload-shim-bower@6.0.3"
    },
})
Does this look right to you as it doesn't seem to pull in the shim? 

Is there an easier way to do this e.g. am I able to change which file is being pulled in by the modules.exports that sits alongside the folder where the package files are contained. In this instance there is a file called ng-file-upload-all.js that includes the shim, it would be great if I could just point to that. If I update it manually it will get overridden when I run a JSPM install on a different computer.

Any help is greatly appreciated

**Comment by guybedford** [comment]
As well as declaring the map, you'll need to add a deps meta config to simulate an `import 'ng-file-upload-shim'` somewhere.

Something like:

```javascript
System.config({
  "meta": {
    "github:danialfarid/ng-file-upload-bower@6.0.3/main-file-path.js": {
      "deps": ["ng-file-upload-shim"]
    }
  }
});
```

Or even skip the map and hard-code the full name in the `deps` above.

**Comment by Quirksmode** [comment]
Massive thanks for getting back to me, I think I understand, is there any further documentation on the meta config I could take a look at?

If I just wanted to target one particular js file within the package (which has the shim built in) how could I use meta for that? Would something like this work?

    System.config({
        "map": {
            "ng-file-upload": "github:danialfarid/ng-file-upload-bower@6.0.3"
        },
     "meta": {
         "github:danialfarid/ng-file-upload-bower@6.0.3/ng-file-upload-all.js": {
             "deps": ["ng-file-upload"]
         }
      }
    })

**Comment by guybedford** [comment]
Yes your example would work. See https://github.com/systemjs/systemjs/blob/master/docs/config-api.md#meta for the SystemJS docs here.

**Comment by Quirksmode** [comment]
Sadly this hasn't worked , not really sure what I am doing wrong :-(

**Comment by guybedford** [comment]
So the config to do what you're trying to do is:

```
System.config({
  "meta": {
    "github:danialfarid/ng-file-upload-bower@6.0.4/ng-file-upload.js": {
      "deps": [
        "ng-file-upload-shim"
      ]
    }
  }
});
```

Note how we have to specify the right file path to the file with deps set.

But what you really need here is just to create an override for these packages to work in jspm. This is the process documented in https://github.com/jspm/registry/wiki/Configuring-Packages-for-jspm.

For examples of overrides for Angular style packages like this see for example https://github.com/jspm/registry/blob/master/package-overrides/github/rzajac/angularjs-slider%400.1.22.json. Note that https://github.com/jspm/registry/blob/master/package-overrides/github/rzajac/angularjs-slider%400.1.22.json#L8 would be `ng-file-upload` to correspond to the main at `ng-file-upload.js` for ng-file-upload-bower.
